### PR TITLE
✉️ feat: out-of-app invite endpoints

### PR DESCRIPTION
**Pull request by stoneliuCS**
# Description

- Created an endpoint where valid manager of a group can fetch a token to share to other users to invite them. 
- Created test cases that cover most (if not all edge cases) 
- Things to note:
- Since generating a new token on every endpoint is a bit tricky, I amended the endpoint to also take in the groupId since it's impossible for different tokens to be decoded and hash to the same value.
- the crypto module is really weird, unsure why it silently fails on hono app request.

## Checklist:

- [x] I have performed a self-review of my code.
- [x] I have added necessary tests (unit, integration, etc.) to cover the new functionality or fix.
- [x] All new and existing tests pass locally.
- [x] I have updated the documentation to reflect changes (e.g., README, code comments).
- [x] I have removed any commented-out code, debug statements, or unnecessary console logs.

### CODE REVIEW

**Comment by stoneliuCS** [comment]
not my best work

File: backend/src/entities/invitations/service.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import { addMemberPayload, GroupInvitation } from "../groups/validator";
+import { InvitationTransaction } from "./transaction";
+
+export interface InvitationService {
+  createInviteToken(groupId: string, userId: string): Promise<GroupInvitation>;
+  verifyInviteToken(token: string, userId: string): Promise<void>;
+}
+
+export class InvitationServiceImpl implements InvitationService {
+  private InvitationTransaction: InvitationTransaction;
```
Comment (in-mai-space) [inline backend/src/entities/invitations/service.ts:17]: tiny nit invitationTransaction is a field so we can do camelCase it!

File: backend/src/tests/groups/group-add-member.test.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    });
+    builder.assertStatusCode(Status.Conflict);
+  });
+
+  it("Should return 404 if the manager requests it", async () => {
+    // Get the token as group manager
+    const builder = await testBuilder.request({
+      app,
+      type: HTTPRequest.PUT,
+      route: `api/v1/groups/verify`,
```
Comment (in-mai-space) [inline backend/src/tests/groups/group-add-member.test.ts:83]: what does this test test for? i think /api/v1/groups/:token/verify is the route, but here the route is api/v1/groups/verify so it will throw 404 cuz the route not found

**Review by in-mai-space — COMMENTED** [review]
🪨🐐 @stoneliuCS  PR LOOKING GOOD, i just have a small nit and also a question about a test case, but it should be pretty good to go!

File: backend/src/tests/groups/group-add-member.test.ts (hunk 2 above)
Comment (stoneliuCS) [reply backend/src/tests/groups/group-add-member.test.ts]: Fixed! my bad, its supposed to test if the manager tries to invite themselves into the group.

File: backend/src/entities/invitations/service.ts (hunk 1 above)
Comment (stoneliuCS) [reply backend/src/entities/invitations/service.ts]: whoops good catch!

**Review by in-mai-space — APPROVED** [review]
LGTM 🐐
